### PR TITLE
Refine assert in Pipeline Model

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]
lgtm

File: dbms/src/Flash/Pipeline/Exec/PipelineExecBuilder.cpp [hunk 1]
```diff
@@ -53,22 +53,22 @@ Block PipelineExecBuilder::getCurrentHeader() const
...
         return source_op->getHeader();
     }
 }
 
 void PipelineExecGroupBuilder::init(size_t init_concurrency)
 {
-    assert(concurrency == 0);
-    assert(init_concurrency > 0);
+    RUNTIME_CHECK(concurrency == 0);
+    RUNTIME_CHECK(init_concurrency > 0);
```
Comment (Lloyd-Pottiger) [inline dbms/src/Flash/Pipeline/Exec/PipelineExecBuilder.cpp:64]: combine them into one?

Comment (ywqzzy) [reply dbms/src/Flash/Pipeline/Exec/PipelineExecBuilder.cpp:64]: > combine them into one?

ok

**Review by xzhangxian1008 — APPROVED** [review]

**Pull request merged**